common/src/main/java/com/emc/mongoose/common/concurrent/DaemonBase.java

```diff
@@ -33,6 +34,9 @@ protected abstract void doClose()
 	public final void start()
 	throws IllegalStateException {
 		if(stateRef.compareAndSet(State.INITIAL, State.STARTED)) {
+			synchronized(state) {
```

Comment:
I have read. But isn't it about notifyAll at the end of synchronized block? Synchronization is inside compareAndSet and notifyAll has no effect at all, isn't it?